packages/cubejs-firebolt-driver/src/FireboltDriver.ts

```diff
@@ -0,0 +1,221 @@
+import {
+  BaseDriver,
+  DriverInterface,
+  StreamTableData,
+  DownloadTableCSVData,
+} from '@cubejs-backend/query-orchestrator';
+import {
+  Firebolt,
+  ConnectionOptions,
+  Connection,
+  OutputFormat,
+} from 'firebolt-sdk';
+import { FireboltQuery } from './FireboltQuery';
+
+export type FireboltDriverConfiguration = {
+  readOnly?: boolean;
+};
+
+const FireboltTypeToGeneric: Record<string, string> = {
+  long: 'bigint'
+};
+
+const COMPLEX_TYPE = /(nullable|array)\((.+)\)/;
+
+export class FireboltDriver extends BaseDriver implements DriverInterface {
+  private readonly connectionParams: ConnectionOptions;
+
+  private readonly config: FireboltDriverConfiguration;
+
+  private readonly firebolt;
+
+  protected connection: Promise<Connection> | null = null;
+
+  public constructor(config: Partial<FireboltDriverConfiguration> = {}) {
+    super(config);
+
+    this.config = {
+      readOnly: true,
+      ...config
+    };
+
+    this.connectionParams = {
+      username: <string>process.env.CUBEJS_DB_USER,
+      password: <string>process.env.CUBEJS_DB_PASS,
+      database: <string>process.env.CUBEJS_DB_NAME,
+      engineEndpoint: <string>process.env.CUBEJS_FIREBOLT_ENGINE_ENDPOINT,
+    };
+
+    const apiEndpoint = process.env.CUBEJS_FIREBOLT_API_ENDPOINT;
```

Comment:
What is the difference between `engineEndpoint` and `apiEndpoint`?

Reply:
`apiEndpoint` is used for authentication (getting auth token). I decided to have it to be able to test on different environments of Firebolt (dev, staging). It is also not required

`engineEndpoint` is a http endpoint for querying database